From 7ef700c96ceeae4526b5bc736aec41b4594853ab Mon Sep 17 00:00:00 2001
From: Murala S Swamy <ssmurala@habana.ai>
Date: Wed, 14 Aug 2024 19:38:06 +0300
Subject: [PATCH] [FSW-6580] fs1: use generic names to lfs functions

With this patch, used generic names to all lfs
functions instead of boot info naming, because
lfs file operations are not only for boot info
and also used for all types of data and variable
storage.
Removed boot info type argument for file name,
so that any other enums can be used for
different type of file naming.

Change-Id: I9338db52a2a88ac19a1b4ccfb352a653e530fe14
---

diff --git a/common/littlefs/lfs_wrapper.c b/common/littlefs/lfs_wrapper.c
index 3366ee5..a644a9d 100644
--- a/common/littlefs/lfs_wrapper.c
+++ b/common/littlefs/lfs_wrapper.c
@@ -5,7 +5,6 @@
  */
 
 #include <hl_boot_desc.h>
-#include <fs1_fw_if.h>
 #include <stdlib.h>
 #include <string.h>
 #include "lfs.h"
@@ -199,28 +198,31 @@
  * Note: Below is logic to convert integer to string.
  * Using standard lib function sprintf causing code size increased.
  */
-#define NUMBER_BASE	10
-char *type2file(enum hl_boot_info_type type)
+#define DECIMAL_NUMBER_BASE	10
+char *type2file(uint32_t type)
 {
 	static char str[LFS_NAME_MAX];
 	int index = 0;
 	int i, j;
-	do {
-        str[index++] = type % NUMBER_BASE + '0';
-        type /= NUMBER_BASE;
-    } while ((type > 0) && (index < LFS_NAME_MAX));
-	if(index < LFS_NAME_MAX)	str[index] = '\0';
 
-    for (i = 0, j = index - 1; i < j; i++, j--) {
-        char temp = str[i];
-        str[i] = str[j];
-        str[j] = temp;
-    }
+	do {
+		str[index++] = type % DECIMAL_NUMBER_BASE + '0';
+		type /= DECIMAL_NUMBER_BASE;
+	} while ((type > 0) && (index < LFS_NAME_MAX));
+
+	if(index < LFS_NAME_MAX)
+		str[index] = '\0';
+
+	for (i = 0, j = index - 1; i < j; i++, j--) {
+		char temp = str[i];
+		str[i] = str[j];
+		str[j] = temp;
+	}
 
 	return str;
 }
 
-static int __boot_info_set(enum hl_boot_info_type type, const void *data,
+static int __lfs_file_set(uint32_t type, const void *data,
 		     uint32_t size, bool is_truncate)
 {
 
@@ -249,15 +251,15 @@
 	}
 }
 
-/* boot_info_set() must use globally allocated 'data' or finalize */
-int boot_info_set(enum hl_boot_info_type type, const void *data,
+/* lfs_file_set() must use globally allocated 'data' or finalize */
+int lfs_file_set(uint32_t type, const void *data,
 		     uint32_t size)
 {
-	return __boot_info_set(type, data, size, true);
+	return __lfs_file_set(type, data, size, true);
 }
 
 /* Copy partial info from boot info data stored for type */
-int boot_info_copy_partial(enum hl_boot_info_type type, void *data,
+int lfs_file_copy_partial(uint32_t type, void *data,
 			uint32_t offset, uint32_t size)
 {
 	lfs_file_t file;
@@ -303,12 +305,12 @@
 	}
 }
 
-int boot_info_copy(enum hl_boot_info_type type, void *desc_data, uint32_t size)
+int lfs_file_copy(uint32_t type, void *desc_data, uint32_t size)
 {
-	return boot_info_copy_partial(type, desc_data, FULL_COPY, size);
+	return lfs_file_copy_partial(type, desc_data, FULL_COPY, size);
 }
 
-uint32_t boot_info_set_empty()
+uint32_t lfs_disk_set_empty()
 {
 	return lfs_init_disk();
 }
diff --git a/common/littlefs/lfs_wrapper.h b/common/littlefs/lfs_wrapper.h
index 413ebad..a5c3116 100644
--- a/common/littlefs/lfs_wrapper.h
+++ b/common/littlefs/lfs_wrapper.h
@@ -10,12 +10,11 @@
 #define BLOCK_SIZE	64 // 32 fails
 
 int lfs_init_disk(void);
-char *type2file(enum hl_boot_info_type type);
 
-int boot_info_set(enum hl_boot_info_type type, const void *data, uint32_t size);
-int boot_info_copy(enum hl_boot_info_type type, void *desc_data, uint32_t size);
-int boot_info_copy_partial(enum hl_boot_info_type type, void *data,
+int lfs_file_set(uint32_t type, const void *data, uint32_t size);
+int lfs_file_copy(uint32_t type, void *desc_data, uint32_t size);
+int lfs_file_copy_partial(uint32_t type, void *data,
 			uint32_t offset, uint32_t size);
-uint32_t boot_info_set_empty();
+uint32_t lfs_disk_set_empty();
 
 #endif	/* _LFS_WRAPPER_H_ */
